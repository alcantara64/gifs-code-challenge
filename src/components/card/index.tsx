import { Button, Card} from "react-bootstrap"
import { GIF } from "../../types";

type AppCard = {
    gif:GIF
    handleView: (gif:GIF)=> void;
}

export const APPCard = ({gif, handleView }:AppCard) => (
<Card style={{ width: '18rem' }}>
  <Card.Img variant="top" src={gif?.images?.fixed_height?.url}  height={300}/>
  <Card.Body>
    <Card.Title title="title">{gif?.title?.length > 20 ? `${gif?.title?.slice(0,20)}...`: gif?.title}</Card.Title>
    <Button onClick={() =>handleView(gif)} style={{width:'100%'}} variant="primary">View</Button>
  </Card.Body>
</Card>)