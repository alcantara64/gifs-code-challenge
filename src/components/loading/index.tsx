import { ReactComponent as Loader } from '../../image/loading.svg';

import './index.css';

const Loading = () => (
  <div id="app-loader">
    <Loader />
  </div>
);

export default Loading;