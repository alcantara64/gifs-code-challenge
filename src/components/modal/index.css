.modal_image {
  min-height: 200;
  min-width: 200;
}
