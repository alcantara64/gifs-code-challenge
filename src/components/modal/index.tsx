import { Button, Modal } from "react-bootstrap";
import { GIF } from "../../types";
import "./index.css";

type ModalProps = {
  show: boolean;
  handleClose: () => void;
  gif: GIF | null;
};

export const AppModal: React.FC<ModalProps> = ({ show, handleClose, gif }) => {
  return (
    <Modal show={show} onHide={handleClose}>
      <Modal.Header closeButton>
        <Modal.Title>{gif?.title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <img
          className="modal_image"
          src={gif?.images.original.url}
          alt={gif?.title}
        />
        <div>{gif?.user?.description}</div>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={handleClose}>
          Close
        </Button>
      </Modal.Footer>
    </Modal>
  );
};
