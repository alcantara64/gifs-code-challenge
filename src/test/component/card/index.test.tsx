import renderer from 'react-test-renderer';
import {APPCard} from '../../../components/card';

it('should render component', () => {
  const mockGif:any = {
    images: {fixed_height :{url:'https://localhost'}},
    title: 'name',

  }

  const component = renderer.create(
    <APPCard gif={mockGif} handleView={(mockGif) => {} } />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});