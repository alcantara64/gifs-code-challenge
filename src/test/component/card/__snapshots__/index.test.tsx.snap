// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    height={300}
    src="https://localhost"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      title="title"
    >
      name
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      type="button"
    >
      View
    </button>
  </div>
</div>
`;
